docs/overview.md

```diff
@@ -0,0 +1,797 @@
+
+There are a number of implied clusters that are not shown in the linkage, but are implicitly referenced by the linkage. These are clusters with only one element: the original inputs. The linkage can be expanded to improve clarity:
+
+```
+[[0. 0. 0. 1.]
+ [1. 1. 0. 1.]
+ [2. 2. 0. 1.]
+ [3. 3. 0. 1.]
+ [0. 1. 1. 2.]
+ [2. 4. 4. 3.]
+ [3. 5. 8. 4.]]
+```
+
+Where the first 4 rows have been added for illustrative purposes and represent clusters of size one, each containing one the input samples. This illustrative linkage can then be understood as follows:
+
+| Index     | Cluster Index 1 | Cluster Index 2 | Distance | Number of Elements | Representation |
+| --------- | --------------- | --------------- | -------- | ------------------ | -------------- |
+| 0         | 0               | 0               | 0        | 1                  | (A)            |
+| 1         | 1               | 1               | 0        | 1                  | (B)            |
+| 2         | 2               | 2               | 0        | 1                  | (C)            |
+| 3         | 3               | 3               | 0        | 1                  | (D)            |
+| 4         | 0               | 1               | 1        | 2                  | (A,B)          |
+| 5         | 2               | 4               | 4        | 3                  | (C,(A,B))      |
+| 6         | 3               | 5               | 8        | 4                  | (D,(C,(A,B)))  |
+
+- Index `0` represents a cluster containing clusters with indices `0` and `0`. It has a distance of `0` and contains `1` element. It represents the cluster: `(A)`.
+- Index `1` represents a cluster containing clusters with indices `1` and `1`. It has a distance of `0` and contains `1` element. It represents the cluster: `(B)`.
+- Index `4` represents a cluster containing clusters with indices `0` and `1`. It has a distance of `1` and contains `2` elements. It represents the cluster: `(A,B)`.
+- Index `5` represents a cluster containing clusters with indices `2` and `4`. It has a distance of `4` and contains `3` elements. It represents the cluster: `(C,(A,B))`.
+- Index `6` represents a cluster containing clusters with indices `3` and `5`. It has a distance of `8` and contains `4` elements. It represents the cluster: `(D,(C,(A,B)))`.
+
+We can see that the hierarchical clustering generated by this linkage `(D,(C,(A,B)))` matches the clustering we generated manually in the previous section `(((A,B),C),D)`. The ordering is reversed, but the clustering and meaning is the same.
+
+### Generating an Address from the Complete-Linkage Hierarchical Clustering
```

Comment:
Can you mention we use scipy `fcluster` here and link to the documentation.

Reply:
I tried to add it where I thought it would flow best: 55e7fa6